testing/python_lib/test_failscale.py

```diff
@@ -59,18 +56,20 @@ def ping_device(device, ping_count):
         ping_counts = {}
         target_args = []
         for device in device_list:
-            ping_count = multiprocessing.Array("i", [0, 1])
+            ping_count = self.get_multiprocessing_array('i', 2)
```

Comment:
not quite what I meant, all you did was rename things, but the _concept_ of multiprocessing is still there -- it's bleeding the abstraction.  Case in point -- what is this multiprocessing array anyway?  Looking at this makes no sense to me (without a detailed understanding of the multiprocessing something something) -- what is i? what is 2?

The real underlying problem here is that the code is not obvious what it's doing. I suppose the other way out of this would be to comment it to explain what it's doing... but in a better world you wouldn't have to do that (documentation is an admission of failure).

I suppose I can't even suggest what the proper fix is because I don't understand what the code is trying to do...  So Maybe start with a comment or two explaining what this array is, and then we can either stick with that or I could suggest a better refactoring.

Reply:
Does it make sense now?